src/include/statistics/backend_stats_context.h

```diff
@@ -109,15 +146,24 @@ class BackendStatsContext {
   std::unordered_map<TableMetric::TableKey, std::unique_ptr<TableMetric>>
       table_metrics_{};
 
-  // Index metrics
+  // TODO stl map is not thread safe. use cuckoomap instead
+  // CuckooMap<IndexMetric::IndexKey,std::shared_ptr<IndexMetric>>
+  //      index_metrics_{};
```

Comment:
Not yet. The CuckooMap container doesn't have begin() and end() iterators, so we can't use it directly to replace stl map, since during aggregation we need to iterate over all elements in the map. We need to come up with some walk around (e.g. use a set to store all the keys, or use a RW lock, which slow down txn performance)